### PR TITLE
fix(mini): dodge status line when cmdheight > 0

### PR DESCRIPTION
**Pull request by Frederick888**
The cmdheight addition in config/init.lua is to avoid breaking the
default setup, where cmdheight is reduced to 0 after MiniView's
initialisation.

---

I'm migrating to LazyVim but I still like https://github.com/gelguy/wilder.nvim.
So I disabled noice.nvim's `cmdline` and `messages`.

But then I noticed that the LSP progress pop-ups started overlapping
with the status line, so here's a patch I managed to put together.

I'm not quite sure if this is the right fix. Feel free to trump this
with a better PR :)

### CODE REVIEW

**Comment by Frederick888** [comment]
Hmmm... When my system is slow, sometimes they can still overlap with my status bar but only when they are disappearing. Suggestions?

**Review by rubiin — APPROVED** [review]

File: lua/noice/config/init.lua [hunk 1]
```diff
@@ -262,6 +262,9 @@ function M.setup(options)
     end,
   })
 
+  if M.options.cmdline.enabled then
+    vim.o.cmdheight = 0
```
Comment (folke) [inline lua/noice/config/init.lua:266]: Why is this needed?

Comment (Frederick888) [reply lua/noice/config/init.lua:266]: If I don't apply it here, I guess due to the fact that MiniView's initialisation happens too early, there's a gap between the pop-ups and `cmdline`.

image

PS: repro.lua
```lua
-- DO NOT change the paths and don't remove the colorscheme
local root = vim.fn.fnamemodify("./.repro", ":p")

-- set stdpaths to use .repro
for _, name in ipairs({ "config", "data", "state", "cache" }) do
  vim.env[("XDG_%s_HOME"):format(name:upper())] = root .. "/" .. name
end

-- bootstrap lazy
local lazypath = root .. "/plugins/lazy.nvim"
if not vim.loop.fs_stat(lazypath) then
  vim.fn.system({ "git", "clone", "--filter=blob:none", "https://github.com/folke/lazy.nvim.git", lazypath, })
end
vim.opt.runtimepath:prepend(lazypath)

-- install plugins
local plugins = {
  "folke/tokyonight.nvim",
  "MunifTanjim/nui.nvim",
  { "https://github.com/Frederick888/noice.nvim.git", branch = 'mini-dodge-status-line' },
  -- add any other plugins here
}
require("lazy").setup(plugins, {
  root = root .. "/plugins",
})

vim.cmd.colorscheme("tokyonight")
-- add anything else here

require("noice").setup({
  lsp = {
    -- override markdown rendering so that **cmp** and other plugins use **Treesitter**
    override = {
      ["vim.lsp.util.convert_input_to_markdown_lines"] = true,
      ["vim.lsp.util.stylize_markdown"] = true,
      ["cmp.entry.get_documentation"] = true, -- requires hrsh7th/nvim-cmp
    },
  },
  -- you can enable a preset for easier configuration
  presets = {
    bottom_search = true, -- use a classic bottom cmdline for search
    command_palette = true, -- position the cmdline and popupmenu together
    long_message_to_split = true, -- long messages will be sent to a split
    inc_rename = false, -- enables an input dialog for inc-rename.nvim
    lsp_doc_border = false, -- add a border to hover docs and signature help
  },

  cmdline = {
    enabled = true,
  },
  messages = {
    enabled = true,
  }
})

local timer = vim.uv.new_timer()
local counter = 1
vim.uv.timer_start(timer, 1000, 1000, function ()
  vim.notify('hello, world! ' .. counter)
  counter = counter + 1
end)
```

**Comment by Frederick888** [comment]
> I just pushed an update that probably also fixes this issue. Can you check?

Unfortunately no. I checked out 7ee3649b9b5f4df5b44ae63ed85b2c9fc834a124 and the issue still persists in both my minimal repro.lua and full setup.

image


repro.lua:

````lua
-- DO NOT change the paths and don't remove the colorscheme
local root = vim.fn.fnamemodify("./.repro", ":p")

-- set stdpaths to use .repro
for _, name in ipairs({ "config", "data", "state", "cache" }) do
  vim.env[("XDG_%s_HOME"):format(name:upper())] = root .. "/" .. name
end

-- bootstrap lazy
local lazypath = root .. "/plugins/lazy.nvim"
if not vim.loop.fs_stat(lazypath) then
  vim.fn.system({ "git", "clone", "--filter=blob:none", "https://github.com/folke/lazy.nvim.git", lazypath, })
end
vim.opt.runtimepath:prepend(lazypath)

-- install plugins
local plugins = {
  "folke/tokyonight.nvim",
  "MunifTanjim/nui.nvim",
  -- { "https://github.com/Frederick888/noice.nvim.git", branch = 'mini-dodge-status-line' },
  "https://github.com/folke/noice.nvim",
  -- add any other plugins here
}
require("lazy").setup(plugins, {
  root = root .. "/plugins",
})

vim.cmd.colorscheme("tokyonight")
-- add anything else here

require("noice").setup({
  lsp = {
    -- override markdown rendering so that **cmp** and other plugins use **Treesitter**
    override = {
      ["vim.lsp.util.convert_input_to_markdown_lines"] = true,
      ["vim.lsp.util.stylize_markdown"] = true,
      ["cmp.entry.get_documentation"] = true, -- requires hrsh7th/nvim-cmp
    },
  },
  -- you can enable a preset for easier configuration
  presets = {
    bottom_search = true, -- use a classic bottom cmdline for search
    command_palette = true, -- position the cmdline and popupmenu together
    long_message_to_split = true, -- long messages will be sent to a split
    inc_rename = false, -- enables an input dialog for inc-rename.nvim
    lsp_doc_border = false, -- add a border to hover docs and signature help
  },

  cmdline = {
    enabled = false,
  },
  messages = {
    enabled = false,
  }
})

local timer = vim.uv.new_timer()
local counter = 1
vim.uv.timer_start(timer, 1000, 1000, function ()
  vim.notify('hello, world! ' .. counter)
  counter = counter + 1
end)

vim.opt.cmdheight = 2
```

**Comment by folke** [comment]
Finally got some time to look into this. This is honestly not a bug.
You can change `opts.views.mini.position = -2`.
That will solve it for your use-case

**Comment by Frederick888** [comment]
@folke Thank you! I didn't know that option existed!

I went through the docs, and for the record, I ended up with

```lua
{
  'folke/noice.nvim',
  opts = {
    cmdline = { enabled = false },
    messages = { enabled = false },
    popupmenu = { enabled = false },
    views = {
      mini = {
        position = {
          row = -3,
        },
      },
    },
  },
},
```
